src/backend/distributed/planner/local_distributed_join_planner.c

```diff
@@ -538,11 +538,6 @@ CreateConversionCandidates(PlannerRestrictionContext *plannerRestrictionContext,
 
 		RelationRestriction *relationRestriction =
 			RelationRestrictionForRelation(rangeTableEntry, plannerRestrictionContext);
```

Comment:
outer joins with WHERE false prevents relation restrictions generated, see https://github.com/citusdata/citus/pull/4292#issue-514909101

And somehow PG decides to do that for this query as `pg_dist_rebalanc_strategy` is not showing up on EXPLAIN output for local-only tables schema. I guess PG decides that `pg_dist_rebalanc_strategy` does not contribute to the result at all and do not generate restrictions

```SQL
explain select 1 from
  pg_operator
  inner join tbl on (select 1 from pg_type) > d
  left join pg_dist_rebalance_strategy on 'a' = name
where a + b + c > 0;
                             QUERY PLAN                             
--------------------------------------------------------------------
 Nested Loop  (cost=19.31..2071.28 rows=158620 width=4)
   InitPlan 1 (returns $0)
     ->  Seq Scan on pg_type  (cost=0.00..19.31 rows=531 width=4)
   ->  Seq Scan on pg_operator  (cost=0.00..21.70 rows=770 width=0)
   ->  Materialize  (cost=0.00..48.03 rows=206 width=0)
         ->  Seq Scan on tbl  (cost=0.00..47.00 rows=206 width=0)
               Filter: (($0 > d) AND (((a + b) + c) > 0))
(7 rows)

```

So, this change makes sense

Reply:
I also guess that the other tests that fail are all expected. Because they are:
```SQL

 SELECT count(*) FROM distributed LEFT JOIN local USING (id);

```

Where, `id` is PRIMARY KEY. So, Postgres doesn't need the inner tables to calculate the result